internal/pkg/task/config_runner.go

```diff
@@ -104,3 +107,65 @@ func (r *ConfigRunner) validateDependencies() error {
 
 	return nil
 }
+
+// CheckNonZeroExitCode returns the status of the containers part of the given tasks.
+func (r *ConfigRunner) CheckNonZeroExitCode(tasks []*Task) error {
+	var essentialContainers []string
+	taskDefName := fmt.Sprintf("copilot-%s", r.GroupName)
+	taskDefinition, err := r.Starter.TaskDefinition(taskDefName)
+	if err != nil {
+		return fmt.Errorf("get task definition %s of a task", taskDefName)
+	}
+
+	for _, container := range taskDefinition.ContainerDefinitions {
+		if aws.BoolValue(container.Essential) {
+			essentialContainers = append(essentialContainers, aws.StringValue(container.Name))
+		}
+	}
+
+	taskARNs := make([]string, len(tasks))
+	for idx, task := range tasks {
+		taskARNs[idx] = task.TaskARN
+	}
+	describedTasks, describeErr := r.Starter.DescribeTasks(r.Cluster, taskARNs)
+	if describeErr != nil {
+		return describeErr
+	}
```

Comment:
Seems like we can make this block running in parallel with https://github.com/aws/copilot-cli/pull/3620/files#diff-f4d4181f6b85ffcf797c84472ae4742e03524f9e77606345fe3c77c2629b9b9dR114-R124

Reply:
Do you think we need to parallelize these two blocks because there is only one call to DescribeTask and TaskDefinition is happening and for loops are also going for max 5 times as there are not much stand-alone tasks that customers deploy and task definition will also have only one container per task. let me know what do you think about it.

Reply:
That's fair. I think we can leave it as is